server/src/main/java/org/elasticsearch/common/bytes/ReleasableBytesReference.java

```diff
@@ -217,6 +220,12 @@ public int arrayOffset() {
         return delegate.arrayOffset();
     }
 
+    @Override
+    public XContentParser xContentParser(XContent xContent, XContentParserConfiguration parserConfig) throws IOException {
+        assert refCount() > 0;
+        return delegate.xContentParser(xContent, parserConfig);
+    }
```

Comment:
I did this here, but something inside me screamed "shouldn't the parser it builds be ref counted too?" But that feels like a problem for another time.